### ISSUE TITLE
[Feature request] Provide the backup service by default.

### ISSUE THREAD

**Issue by cobenash** [mattermost/docker#61]
## Motivation/Problems
The current structure doesn't provide the backup service. Therefore, when we faces the database broken or something unpredictable, the maintainer couldn't restore it back easily.

## Proposed resolution
We could use some useful tools like below which provides backup for Postgres and easily backup to S3.
* https://github.com/appleboy/docker-backup-database

**Comment by almereyda** [comment]
I think this kind of maintenance setup is out of scope for a Docker example. As said  in https://github.com/mattermost/docker/issues/74#issuecomment-994156202:

> (please have common sense in place, and a good backup!)

A backup is rather a feature of the underlying deployment platform, and cannot be taken into account from each deployment individually. As is a hard requirement on S3 or local disk stores, which may not be available in all environments.

Secondly, the project that you suggest seems low on maintenance, as there is only a single person that is responsible for it. (https://github.com/appleboy/docker-backup-database/graphs/contributors)

Should we rather rephrase this issue, in so it covers to document suitable backup strategies, like the one suggested by you? We could begin to populate the GitHub wiki here, and invite others to do the same.

For a production-ready backup of an instance, next to the database, I would also suggest to keep a copy of the `volumes` directory, which also contains state of the application.

**Comment by cobenash** [comment]
@almereyda 

Yeah, I agree with you. It sounds like a good plan that we write a WIKI or some document of backup strategies.
